src/Angor/Shared/Services/MempoolSpaceIndexerApi.cs

```diff
@@ -414,6 +416,53 @@ public async Task<string> GetTransactionHexByIdAsync(string transactionId)
         };
     }
 
+    private async Task PopulateMissingSpentDataIfAny(List<Outspent> outspents, MempoolTransaction mempoolTransaction)
+    {
+        var indexer = _networkService.GetPrimaryIndexer();
+
+        for (int index = 0; index < outspents.Count; index++)
+        {
+            if (outspents[index].Spent)
+            {
+                if (outspents[index].Txid == null)
+                {
+                    var output = mempoolTransaction.Vout[index];
+                    if (output != null && !string.IsNullOrEmpty(output.ScriptpubkeyAddress))
+                    {
+                        var txsUrl = $"{MempoolApiRoute}/address/{output.ScriptpubkeyAddress}/txs";
+
+                        var response = await _httpClient.GetAsync(indexer.Url + txsUrl);
+                        _networkService.CheckAndHandleError(response);
+
+                        if (!response.IsSuccessStatusCode)
+                            throw new InvalidOperationException(response.ReasonPhrase);
+
+                        var trx = await response.Content.ReadFromJsonAsync<List<MempoolTransaction>>(new JsonSerializerOptions()
+                            { PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower });
+
+                        bool found = false;
+                        foreach (var transaction in trx)
```

Comment:
I think it's better to first find the transaction and than iterate through the inputs in it, you go though thousands of transactions and you go through every vin in them before reaching your data for no reason.

Reply:
ideally yes but I realize the trx id to compare against is inside the input so I anyway have to iterate each vin inside each trx.